### PR TITLE
allow --change-address to be omitted and default to internal wallet address

### PR DESCRIPTION
**Pull request by Einliterflasche**
Closes https://github.com/comit-network/xmr-btc-swap/issues/1696

### CODE REVIEW

**Comment by Einliterflasche** [comment]
This change isn't as straight forward as one would think it would be because the the `structopt` proc macro doesn't seem to work as intended.

File: swap/src/api.rs [hunk 1]
```diff
@@ -267,6 +267,13 @@ impl Context {
             tasks: Arc::new(PendingTaskList::default()),
         }
     }
+
+    pub fn bitcoint_wallet(&self) -> Option<Arc<bitcoin::Wallet>> {
```
Comment (binarybaron) [inline swap/src/api.rs:271]: Typo

**Review by binarybaron — CHANGES_REQUESTED** [review]

**Comment by Einliterflasche** [comment]
The flag is now added, but the help dialogue for `swap buy-xmr` now shows:

```rust
Error: error: The following required arguments were not provided:
    --change-address <bitcoin-change-address>
    --receive-address <monero-receive-address>
    --refund-to-internal-wallet
    --seller <seller>

USAGE:
    swap buy-xmr --change-address <bitcoin-change-address> --receive-address <monero-receive-address> --refund-to-internal-wallet --seller <seller> --tor-socks5-port <tor-socks5-port>
```

even though the command works as intended (with `--refund-to-internal-wallet`, `--change-address` isn't required). Using both flags correctly gives `Error: error: The argument '--change-address <bitcoin-change-address>' cannot be used with '--refund-to-internal-wallet'`

**Comment by Einliterflasche** [comment]
lmao did you just merge into my branch?

**Comment by Einliterflasche** [comment]
w8 i still have to test 💀

**Comment by Einliterflasche** [comment]
I added a logging message and tested the flag. Do you think we need a unit test? I don't really think so.

**Comment by delta1** [comment]
Does this require a flag? change-address could be optional, and if it's excluded then we can default to using the internal wallet (with a log message and/or confirmation)

**Comment by delta1** [comment]
Yep I think what I've suggested is a better UX and should just be communicated to the user, and documented in the changelog/docs. It's actually always felt painful to have to supply a change address.

**Comment by Einliterflasche** [comment]
Very well. It seems that clippy agrees with you ;) 

```rust
error: field `refund_to_internal_wallet` is never read
   --> swap/src/cli/command.rs:336:9
    |
314 |     BuyXmr {
    |     ------ field in this variant
...
336 |         refund_to_internal_wallet: bool,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `-D dead-code` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(dead_code)]`
```

I'll push what you suggested shortly.

**Comment by delta1** [comment]
Thanks @Einliterflasche ! Hopefully @binarybaron agrees :)

**Comment by binarybaron** [comment]
> Yep I think what I've suggested is a better UX and should just be communicated to the user, and documented in the changelog/docs. It's actually always felt painful to have to supply a change address.

I agree 👍

File: swap/src/cli/command.rs [hunk 2]
```diff
@@ -69,6 +69,36 @@ where
...
+                None => {
+                    let internal_wallet = context
+                        .bitcoin_wallet()
+                        .expect("bitcoin wallet should exist")
+                        .new_address()
+                        .await?;
+
+                    tracing::info!(
+                        internal_wallet=%internal_wallet,
+                        "No --change-address supplied. Incase of a refund funds will be sent to internal wallet."
```
Comment (delta1) [inline swap/src/cli/command.rs:95]:
```suggestion
                        "No --change-address supplied. Any change will be received to the internal wallet."
```

**Comment by delta1** [comment]
Please update the changelog and any docs that refer to change-address

**Comment by binarybaron** [comment]
LGTM!

**Pull request merged**